### PR TITLE
Add Oxygen bar

### PR DESCRIPTION
**Pull request by samuelWilliams99**
image

It go down when under water - wow

### CODE REVIEW

**Review by plally — APPROVED** [review]

File: gamemode/modules/drowning/cl_drowning.lua [hunk 1]
```diff
@@ -17,7 +20,28 @@ net.Receive( "RVR_Player_Take_Drown_Damage", function()
...
-    -- TODO: Add HUD elements for drowning
+    if not shouldDrawDrowningElement then return end
+
+    local timePassed = CurTime() - drownTimeStart
+    local prog = timePassed / GAMEMODE.Config.Drowning.DROWNING_THRESHOLD
+
+    local hotbarFrame = RVR.Inventory.hotbar.frame
+    local x, y = hotbarFrame:GetPos()
+    local w = hotbarFrame:GetWide()
+    local wToRemove = 40
```
Comment (overlordbleck) [inline gamemode/modules/drowning/cl_drowning.lua]: padding or margin here?

File: gamemode/modules/drowning/sv_drowning.lua [hunk 2]
```diff
@@ -8,35 +8,36 @@ util.AddNetworkString( "RVR_Player_Enter_Water" )
 util.AddNetworkString( "RVR_Player_Leave_Water" )
 util.AddNetworkString( "RVR_Player_Take_Drown_Damage" )
 
-local function isDrowning( player )
-    return ( CurTime() - player.FirstSubmergedTime ) >= config.DROWNING_THRESHOLD
+local function isDrowning( ply )
```
Comment (overlordbleck) [inline gamemode/modules/drowning/sv_drowning.lua:11]: What, you don't like `player`?

File: gamemode/modules/drowning/sv_drowning.lua [hunk 3]
```diff
@@ -56,6 +57,9 @@ local function drowningCheck()
                 takeDrownDamage( ply )
             end
         else
+            if ply.IsInWater then
```
Comment (overlordbleck) [inline gamemode/modules/drowning/sv_drowning.lua:60]: good catch :-)

File: gamemode/modules/drowning/sv_drowning.lua (hunk 2 above)
Comment (samuelWilliams99) [reply gamemode/modules/drowning/sv_drowning.lua:11]: Its an existing gmod library

Comment (samuelWilliams99) [reply gamemode/modules/drowning/sv_drowning.lua:11]: https://wiki.facepunch.com/gmod/player(library)

**Review by overlordbleck — APPROVED** [review]

**Pull request merged**